--- lib/IO/Async/Routine.pm.orig
+++ lib/IO/Async/Routine.pm
@@ -214 +214 @@
-      die "No viable Routine models";
+      undef;
@@ -221,0 +222 @@
+   defined $params->{model} or croak "No viable Routine models";
--- t/14loop-poll-process.t.orig
+++ t/14loop-poll-process.t
@@ -6,0 +7,2 @@
+use IO::Async::OS;
+plan('skip_all' => "POSIX fork() is not available") unless IO::Async::OS->HAVE_POSIX_FORK;
--- t/14loop-select-process.t.orig
+++ t/14loop-select-process.t
@@ -6,0 +7,2 @@
+use IO::Async::OS;
+plan('skip_all' => "POSIX fork() is not available") unless IO::Async::OS->HAVE_POSIX_FORK;
--- t/26pid.t.orig
+++ t/26pid.t
@@ -14,0 +15,2 @@
+use IO::Async::OS;
+plan('skip_all' => "POSIX fork() is not available") unless IO::Async::OS->HAVE_POSIX_FORK;
--- t/41routine.t.orig
+++ t/41routine.t
@@ -8,0 +9,2 @@
+use IO::Async::OS;
+plan('skip_all' => "No viable Routine models") unless IO::Async::OS->HAVE_POSIX_FORK || IO::Async::OS->HAVE_THREADS;
--- t/42function.t.orig
+++ t/42function.t
@@ -16,0 +17 @@
+plan('skip_all' => "No viable Routine models") unless IO::Async::OS->HAVE_POSIX_FORK || IO::Async::OS->HAVE_THREADS;
--- t/50resolver.t.orig
+++ t/50resolver.t
@@ -10,0 +11,2 @@
+use IO::Async::OS;
+plan('skip_all' => "No viable Routine models") unless IO::Async::OS->HAVE_POSIX_FORK || IO::Async::OS->HAVE_THREADS;
--- t/52loop-listen.t.orig
+++ t/52loop-listen.t
@@ -8,0 +9,2 @@
+use IO::Async::OS;
+plan('skip_all' => "No viable Routine models") unless IO::Async::OS->HAVE_POSIX_FORK || IO::Async::OS->HAVE_THREADS;
--- t/70future-io.t.orig
+++ t/70future-io.t
@@ -68,10 +68,11 @@
-run_tests qw(
+my @future_io_tests = qw(
    sleep
    poll_no_hup
    read sysread write syswrite
    connect accept
    send recv
-   waitpid
 );
+push @future_io_tests, "waitpid" if IO::Async::OS->HAVE_POSIX_FORK;
+run_tests @future_io_tests;
 
 done_testing;
