--- lib/XML/TreePP.pm.orig
+++ lib/XML/TreePP.pm
@@ -549,6 +549,11 @@
     }
 
     my $text = $self->hash_to_xml( undef, $tree );
+    if ( $ALLOW_UTF8_FLAG
+        && ! &_tree_has_utf8_flag($tree)
+        && utf8::is_utf8($text) ) {
+        utf8::downgrade($text, 1);
+    }
     if ( $from && $to ) {
         my $stat = $self->encode_from_to( \$text, $from, $to );
         return $self->die( "Unsupported encoding: $to" ) unless $stat;
@@ -864,6 +869,28 @@
     $flat;
 }
 
+sub _tree_has_utf8_flag {
+    my $value = shift;
+    if ( ref $value ) {
+        if ( UNIVERSAL::isa( $value, 'HASH' ) ) {
+            foreach my $key ( keys %$value ) {
+                return 1 if utf8::is_utf8($key);
+                return 1 if &_tree_has_utf8_flag( $value->{$key} );
+            }
+        }
+        elsif ( UNIVERSAL::isa( $value, 'ARRAY' ) ) {
+            foreach my $item (@$value) {
+                return 1 if &_tree_has_utf8_flag($item);
+            }
+        }
+        elsif ( UNIVERSAL::isa( $value, 'SCALAR' ) ) {
+            return utf8::is_utf8($$value);
+        }
+        return;
+    }
+    return utf8::is_utf8($value);
+}
+
 sub flat_to_tree {
     my $self   = shift;
     my $source = shift;
