--- perl5/cpan/Archive-Tar/lib/Archive/Tar/Constant.pm
+++ src/main/perl/lib/Archive/Tar/Constant.pm
@@ -75,8 +75,7 @@
 
                             ### allow BZIP to be turned off using ENV: DEBUG only
 use constant BZIP           => do { !$ENV{'PERL5_AT_NO_BZIP'} and
-                                        eval { require IO::Uncompress::Bunzip2;
-                                               require IO::Compress::Bzip2; };
+                                        eval { require Compress::Bzip2; };
                                     $ENV{'PERL5_AT_NO_BZIP'} || $@ ? 0 : 1
                                 };
 
