--- perl5/cpan/IPC-Cmd/lib/IPC/Cmd.pm
+++ src/main/perl/lib/IPC/Cmd.pm
@@ -8,5 +8,9 @@
     use constant IS_HPUX        => $^O eq 'hpux'                      ? 1 : 0;
     use constant IS_WIN98       => (IS_WIN32 and !Win32::IsWinNT())   ? 1 : 0;
+    use constant IS_PERLONJAVA  => do {
+        require Config;
+        $Config::Config{perlonjava} ? 1 : 0;
+    };
     use constant ALARM_CLASS    => __PACKAGE__ . '::TimeOut';
     use constant SPECIAL_CHARS  => qw[< > | &];
     use constant QUOTE          => do { IS_WIN32 ? q["] : q['] };
@@ -38,7 +42,7 @@
         require Time::HiRes; Time::HiRes->import();
         require Win32 if IS_WIN32;
     };
-    $CAN_USE_RUN_FORKED = $@ || !IS_VMS && !IS_WIN32;
+    $CAN_USE_RUN_FORKED = !IS_PERLONJAVA && ($@ || !IS_VMS && !IS_WIN32);
 
     eval {
         my $wait_start_time = Time::HiRes::clock_gettime(&Time::HiRes::CLOCK_MONOTONIC);
