--- perl5/t/test.pl
+++ t/test.pl
@@ -1,3 +1,7 @@
+# --------------------------------------------
+# Modified t/test.pl for running Perl test suite with PerlOnJava
+# --------------------------------------------
+
 #
 # t/test.pl - most of Test::More functionality without the fuss
 
@@ -587,16 +591,16 @@
     last SKIP;
 }
 
 sub skip_if_miniperl {
-    skip(@_) if is_miniperl();
+    ## PerlOnJava is not miniperl
+    # skip(@_) if is_miniperl();
 }
 
 sub skip_without_dynamic_extension {
-    my $extension = shift;
-    skip("no dynamic loading on miniperl, no extension $extension", @_)
-	if is_miniperl();
-    return if &_have_dynamic_extension($extension);
-    skip("extension $extension was not built", @_);
+    ## PerlOnJava has dynamic extension
+    # my $extension = shift;
+    # skip("no dynamic loading on miniperl, no extension $extension", @_)
+	# if is_miniperl();
+    # return if &_have_dynamic_extension($extension);
+    # skip("extension $extension was not built", @_);
 }
 
 sub todo_skip {
