--- lib/Graph/AdjacencyMap.pm.orig
+++ lib/Graph/AdjacencyMap.pm
@@ -75,19 +75,25 @@
 		? map [$_, join '|', map "@$_", @$_], @p
 		: map [$_,"@$_"], @p; # use the Schwartz
     if ($arity == 2) {
-	require Set::Object;
-	my ($pre, $suc, @s) = (Set::Object->new(map $_->[0], @p), Set::Object->new(map $_->[1], @p));
-	@rows = ([ 'to:', @s = sort $suc->members ], map {
-	    my $p = $_;
-	    [ $p, map {
-		my $text = defined(my $id = $m->has_path([$p, $_])) ? 1 : '';
+	my (%pre, %suc, @s);
+	for my $path (@p) {
+	    $pre{ $path->[0] } = 1;
+	    $suc{ $path->[1] } = 1;
+	}
+	@rows = ([ 'to:', @s = sort keys %suc ]);
+	for my $p (sort keys %pre) {
+	    my @row = ($p);
+	    for my $s (@s) {
+		my $text = defined(my $id = $m->has_path([$p, $s])) ? 1 : '';
 		my $attrs = !$text ? undef :
-		    $multi ? $m->[ _attr ][$id] : $m->_get_path_attrs([$p, $_]);
-		defined $attrs ? $m->_dumper($attrs) : $text;
-	    } @s ];
-	} sort $pre->members);
+		    $multi ? $m->[ _attr ][$id] : $m->_get_path_attrs([$p, $s]);
+		push @row, defined $attrs ? $m->_dumper($attrs) : $text;
+	    }
+	    push @rows, \@row;
+	}
     } else {
 	@rows = map {
+	    my $p = $_;
 	    my $attrs = $multi
 		? $m->[ _attr ][ $m->has_path($_) ] : $m->_get_path_attrs($_);
 	    [ $m->_dumper($_),
